workflow/controller/operator.go

```diff
@@ -1265,9 +1255,47 @@ func (woc *wfOperationCtx) assessNodeStatus(pod *apiv1.Pod, node *wfv1.NodeStatu
 	return nil
 }
 
-func getExitCode(pod *apiv1.Pod) *int32 {
+func (woc *wfOperationCtx) updateNodeExitData(pod *apiv1.Pod, node *wfv1.NodeStatus) *wfv1.NodeStatus {
+	updated := false
+	containerName := common.MainContainerName
+	if node.Type == wfv1.NodeTypeContainer {
+		containerName = node.DisplayName
```

Comment:
interesting - I think this is a work-around - we should use some shared func to determine the container name - at least so the func acts are docs